src/rank_llm/demo/rerank_inline_docs.py

```diff
@@ -18,14 +16,6 @@
     "Capital punishment (the death penalty) has existed in the United States since before the United States was a country. As of 2017, capital punishment is legal in 30 of the 50 states.",
 ]
 
-results = retrieve_and_rerank(
-    model_path="castorini/rank_zephyr_7b_v1_full",
-    dataset=docs,
-    retrieval_mode=RetrievalMode.QUERY_AND_DOCUMENTS,
-    retrieval_method=RetrievalMethod.UNSPECIFIED,
-    top_k_candidates=len(docs),
-    print_prompts_responses=True,
-    query=query,
-    variable_passages=True,
-    system_message="You are RankLLM, an intelligent assistant that can rank passages based on their relevancy to the query.",
-)
+retrieved_results = Retriever.from_inline_documents(query, documents=docs)

```

Comment:
is `retrieved_results` going to be it's own class? if so, then we can just use a basic constructor?
e.g., `results = Result(query, docs)`

Reply:
hm, I am not sue if I am following:
retriever will return an array of objects that reranker will use. Currently this is a list of dictionaries (dumped as json when storing in file), I am working on creating a result class so that the return value here will be a list[Result].
If your suggestion is for the result class to call the retriever, I'd prefer to keep it this way. Since `result` class is a lightweight class to give some structure to the otherwise vague dictionary, look at it as a dataclass. This will also be the output of the reranker as we spoke. IMO, It should not call retrieve or rerank.
```
class RankingExecInfo:
    def __init__(
        self, prompt, response: str, input_token_count: int, output_token_count: int
    ) -> None:
        self.prompt = prompt
        self.response = response
        self.input_token_count = input_token_count
        self.output_token_count = output_token_count


class Result:
    def __init__(
        self,
        query: str,
        hits: List[Dict[str, Any]],
        ranking_exec_summary: List[RankingExecInfo] = None,
    ):
        self.query = query
        self.hits = hits
        self.ranking_exec_summary = ranking_exec_summary


class ResultsWriter:
    def __init__(self, results: List[Result], append: bool = False):
        self._results = results
        self._append = append

    def write_ranking_exec_summary(self, filename: str):
    ....
```